code/datums/flock/flock.dm

```diff
@@ -202,19 +201,96 @@
+/datum/flock/proc/spawnTrace()
+	var/mob/living/intangible/flock/trace/T = new(usr.loc, src)
+	return T
+
 /datum/flock/proc/addTrace(var/mob/living/intangible/flock/trace/T)
 	if(!T)
 		return
 	src.traces |= T
 	var/datum/abilityHolder/flockmind/aH = src.flockmind.abilityHolder
-	aH.updateCompute()
+	aH?.updateCompute()
 
 /datum/flock/proc/removeTrace(var/mob/living/intangible/flock/trace/T)
 	if(!T)
 		return
 	src.traces -= T
 	var/datum/abilityHolder/flockmind/aH = src.flockmind.abilityHolder
-	aH.updateCompute()
+	aH?.updateCompute()
+
+/datum/flock/proc/ping(var/atom/target, var/mob/living/intangible/flock/pinger)
+	//awful typecheck because turfs and movables have vis_contents defined seperately because god hates us
+	if (!istype(pinger) || (!istype(target, /atom/movable) && !istype(target, /turf)))
+		return
+
+	target.AddComponent(/datum/component/flock_ping)
+
+	for (var/mob/living/intangible/flock/F in (src.traces + src.flockmind))
+		if (F != pinger)
+			var/image/arrow = image(point_img, loc = F, layer = HUD_LAYER)
+			arrow.color = "#00ff9dff"
+			arrow.pixel_y = 20
+			arrow.transform = matrix(arrow.transform, 2,2, MATRIX_SCALE)
+			var/angle = 180 + get_angle(F, target)
+			arrow.transform = matrix(arrow.transform, angle, MATRIX_ROTATE)
+			F.client?.images += arrow
+			animate(arrow, time = 3 SECONDS, alpha = 0)
+			SPAWN(3 SECONDS)
+				F.client?.images -= arrow
+				qdel(arrow)
+		var/class = "flocksay ping [istype(F, /mob/living/intangible/flock/flockmind) ? "flockmindsay" : ""]"
+		var/prefix = "<span class='bold'>\[[src.name]\] </span><span class='name'>[pinger.name]</span>"
+		boutput(F, "<span class='[class]'><a href='?src=\ref[F];origin=\ref[target];ping=[TRUE]'>[prefix]: Interrupt request, target: [target] in [get_area(target)].</a></span>")
+	playsound_global(src.traces + src.flockmind, "sound/misc/flockmind/ping.ogg", 50, 0.5)
+
+//is this a weird use case for components? probably, but it's kinda neat
+/datum/component/flock_ping
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	var/const/duration = 5 SECOND
+	var/end_time = -1
+	var/obj/dummy = null
+
+	Initialize()
+		//initialize is called regardless of whether the component is actually added or not
+		if (parent.GetExactComponent(/datum/component/flock_ping))
+			return
+		//this cast is horribly unsafe, but we just need vis_contents
+		var/atom/movable/target = parent
+
+		src.end_time = world.timeofday + duration
+
+		dummy = new()
+		dummy.layer = target.layer
+		dummy.plane = PLANE_FLOCKVISION
+		dummy.invisibility = INVIS_FLOCKMIND
+		dummy.appearance_flags = PIXEL_SCALE | RESET_TRANSFORM | RESET_COLOR | PASS_MOUSE
+		dummy.icon = target.icon
+		dummy.icon_state = target.icon_state
+		target.render_target = ref(parent)
+		dummy.render_source = target.render_target
+		dummy.add_filter("outline", 1, outline_filter(size=1,color="#00ff9d"))
+		target.vis_contents += dummy
+
+		SPAWN(0)
+			while(world.timeofday < src.end_time)
```

Comment:
TIME
currently will lead to lots of !FUN! at midnight.

Reply:
!FUN! removed